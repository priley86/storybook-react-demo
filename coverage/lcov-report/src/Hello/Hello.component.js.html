<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Hello/Hello.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/Hello</a> Hello.component.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">11.11% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/27</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">10% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">11.11% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/27</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styles from './Hello.shadow.css'
&nbsp;
/**
 * Cloning contents from a &lt;template&gt; element is more performant
 * than using innerHTML because it avoids addtional HTML parse costs.
 */
const helloTemplate = document.createElement('template')
helloTemplate.innerHTML = `&lt;button&gt;&lt;slot name="content"&gt;&lt;/slot&gt;&lt;/button&gt;`
&nbsp;
/**
 * Hello Web Component defines a custom button and its
 * encapsulated Shadow DOM styles
 * 
 * Usage: &lt;hello-component disabled&gt;Hello&lt;/hello-component&gt;
 */
export default class HelloComponent extends HTMLElement {
  /*
  * An instance of the element is created or upgraded
  */
<span class="fstat-no" title="function not covered" >  co</span>nstructor() {
<span class="cstat-no" title="statement not covered" >    super()</span>
<span class="cstat-no" title="statement not covered" >    this.attachShadow({ mode: 'open' })</span>
<span class="cstat-no" title="statement not covered" >    this.shadowRoot.appendChild(helloTemplate.content.cloneNode(true))</span>
  }
&nbsp;
  /**
   * `connectedCallback` hooks up the even listeners and styles the component
   */
<span class="fstat-no" title="function not covered" >  co</span>nnectedCallback() {
<span class="cstat-no" title="statement not covered" >    this._styleComponent()</span>
<span class="cstat-no" title="statement not covered" >    this.addEventListener('click', this._onClick)</span>
  }
&nbsp;
  static get observedAttributes() {
    return ['disabled', 'class']
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  at</span>tributeChangedCallback(attr, oldVal, newVal) {
<span class="cstat-no" title="statement not covered" >    this[attr] = newVal</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  _s</span>tyleComponent() {
    // Create a style template with our element's shadow css
    const styleTemplate = <span class="cstat-no" title="statement not covered" >document.createElement('template')</span>
<span class="cstat-no" title="statement not covered" >    styleTemplate.innerHTML = `&lt;style&gt;${styles}&lt;/style&gt;`</span>
&nbsp;
    /*
      ShadyCSS is the polyfill for Shadow DOM that we need to activate 
      for each template in browsers that need it
    */
<span class="cstat-no" title="statement not covered" >    if (!window.ShadyCSS.nativeShadow) {</span>
      // ShadyCSS will rename classes as needed to ensure style scoping.
<span class="cstat-no" title="statement not covered" >      window.ShadyCSS.prepareTemplate(styleTemplate, 'hello-component')</span>
<span class="cstat-no" title="statement not covered" >      window.ShadyCSS.styleElement(this)</span>
    } else {
<span class="cstat-no" title="statement not covered" >      this.shadowRoot.appendChild(styleTemplate.content.cloneNode(true))</span>
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  _o</span>nClick() {
<span class="cstat-no" title="statement not covered" >    this.dispatchEvent(new CustomEvent('hello-component.onClick', {}))</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ge</span>t disabled() {
<span class="cstat-no" title="statement not covered" >    return this._disabled</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  se</span>t disabled(val) {
<span class="cstat-no" title="statement not covered" >    if (this.hasAttribute('disabled')) {</span>
<span class="cstat-no" title="statement not covered" >      this._disabled = true</span>
<span class="cstat-no" title="statement not covered" >      this.shadowRoot.querySelector('button').className = 'btn-disabled'</span>
    } else {
<span class="cstat-no" title="statement not covered" >      this._disabled = false</span>
<span class="cstat-no" title="statement not covered" >      this.shadowRoot.querySelector('button').className = this._class</span>
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ge</span>t class() {
<span class="cstat-no" title="statement not covered" >    return this._class</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  se</span>t class(val) {
<span class="cstat-no" title="statement not covered" >    if (this._class === val || this._disabled) {</span>
<span class="cstat-no" title="statement not covered" >      return</span>
    }
    // set the button class on our button along with the _class property
<span class="cstat-no" title="statement not covered" >    this.shadowRoot.querySelector('button').className = val</span>
<span class="cstat-no" title="statement not covered" >    this._class = val</span>
  }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Oct 14 2017 01:21:16 GMT-0400 (EDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
