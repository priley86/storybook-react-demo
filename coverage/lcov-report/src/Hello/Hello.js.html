<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/Hello/Hello.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/Hello</a> Hello.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/14</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import PropTypes from 'prop-types'
import { HelloComponent } from '../index.js'
&nbsp;
// globally define the HelloComponent in the custom element registry
window.customElements.define('hello-component', HelloComponent)
&nbsp;
/**
 * Hello Component React wrapper
 */
class Hello extends React.Component {
  componentDidMount() {
    const { buttonClicked, disabled } = this.props
    <span class="missing-if-branch" title="if path not taken" >I</span>if (this.component) {
<span class="cstat-no" title="statement not covered" >      this.setDisabled(disabled)</span>
<span class="cstat-no" title="statement not covered" >      this.component.addEventListener('hello-component.onClick', <span class="fstat-no" title="function not covered" >e </span>=&gt; {</span>
<span class="cstat-no" title="statement not covered" >        if (buttonClicked) {</span>
<span class="cstat-no" title="statement not covered" >          buttonClicked(e)</span>
        }
      })
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentDidUpdate() {
<span class="cstat-no" title="statement not covered" >    if (this.component) {</span>
<span class="cstat-no" title="statement not covered" >      this.setDisabled(this.props.disabled)</span>
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  se</span>tDisabled(disabled) {
<span class="cstat-no" title="statement not covered" >    disabled</span>
      ? this.component.setAttribute('disabled', '')
      : this.component.removeAttribute('disabled')
  }
&nbsp;
  render() {
    const { className, children } = this.props
&nbsp;
    // React 16 allows custom attributes (such as class)
    // https://reactjs.org/blog/2017/09/08/dom-attributes-in-react-16.html
    // Spread attributes in React 15 no longer needed:
    // let customAttributes = {
    //   'class': className,
    // }
    // return ( &lt;hello-component {...customAttributes}&gt;&lt;/hello-component&gt;
    return (
      &lt;hello-component
        ref={c =&gt; {
          this.component = c
        }}
        class={className}
      &gt;
        {children}
      &lt;/hello-component&gt;
    )
  }
}
&nbsp;
Hello.propTypes = {
  /** callback when button clicked */
  buttonClicked: PropTypes.func,
  /** className */
  className: PropTypes.string,
  /** button disabled */
  disabled: PropTypes.bool,
  /** children nodes  */
  children: PropTypes.node
}
export default Hello
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Oct 14 2017 01:21:16 GMT-0400 (EDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
