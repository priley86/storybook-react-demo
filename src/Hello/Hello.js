import React from 'react'
import PropTypes from 'prop-types'
import { HelloComponent } from '../index.js'

// globally define the HelloComponent in the custom element registry
window.customElements.define('hello-component', HelloComponent)

/**
 * Hello Component React wrapper
 */
class Hello extends React.Component {
  componentDidMount() {
    const { buttonClicked, disabled } = this.props
    if (this.component) {
      this.setDisabled(disabled)
      this.component.addEventListener('hello-component.onClick', e => {
        if (buttonClicked) {
          buttonClicked(e)
        }
      })
    }
  }

  componentDidUpdate() {
    if (this.component) {
      this.setDisabled(this.props.disabled)
    }
  }

  setDisabled(disabled) {
    disabled
      ? this.component.setAttribute('disabled', '')
      : this.component.removeAttribute('disabled')
  }

  render() {
    const { className, children } = this.props

    // React 16 allows custom attributes (such as class)
    // https://reactjs.org/blog/2017/09/08/dom-attributes-in-react-16.html
    // Spread attributes in React 15 no longer needed:
    // let customAttributes = {
    //   'class': className,
    // }
    // return ( <hello-component {...customAttributes}></hello-component>
    return (
      <hello-component
        ref={c => {
          this.component = c
        }}
        class={className}
      >
        {children}
      </hello-component>
    )
  }
}

Hello.propTypes = {
  /** callback when button clicked */
  buttonClicked: PropTypes.func,
  /** className */
  className: PropTypes.string,
  /** button disabled */
  disabled: PropTypes.bool,
  /** children nodes  */
  children: PropTypes.node
}
export default Hello
