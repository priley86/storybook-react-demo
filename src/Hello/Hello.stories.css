html {
  --accent-background-color: #ff4081;
}
