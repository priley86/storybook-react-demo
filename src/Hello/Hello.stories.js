import React from 'react'
import classnames from 'classnames'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs'
import { withInfo } from '@storybook/addon-info'
import { Hello } from '../index'
import './Hello.stories.css'

const stories = storiesOf('Hello', module)
stories.addDecorator(withKnobs)

stories.add(
  'with slot styles',
  withInfo('the hello component with slot styles')(() => {
    const slotText = text('Slot', 'Hello Shadow Dom!')
    const disabled = boolean('Disabled', false)
    const buttonClass = select(
      'Button Class',
      ['btn-primary', 'btn-accent', 'btn-warn'],
      'btn-primary'
    )
    const bold = boolean('Bold', false)

    // React 15 requires a ref to set slot attributes, you can use this helper for React 15...
    // https://github.com/skatejs/skatejs/issues/1096
    // const slot = (name = '') => { return {
    //     ref: e => { e.setAttribute('slot', name)}
    //   }
    // }
    return (
      <div>
        <Hello
          buttonClicked={action('button clicked')}
          className={buttonClass}
          disabled={disabled}
        >
          {/** React 16 now natively supports slots and custom attributes so this is not needed **/}
          {/** <span {...slot('content')}>{slotText}</span> **/}
          <span slot="content" className={classnames({ bold: bold })}>
            {slotText}
          </span>
        </Hello>
      </div>
    )
  })
)
