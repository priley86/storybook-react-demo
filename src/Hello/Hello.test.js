/* eslint-env jest */
/* 
  JS Dom Web Component Polyfill Workaround for Jest:
  https://github.com/tmpvar/jsdom/issues/1030
  Will be resolved in: 
  https://github.com/tmpvar/jsdom/pull/1872
*/
import 'document-register-element'

import React from 'react'
import renderer from 'react-test-renderer'
import { default as Hello } from './Hello.js'

test('Hello renders default properly', () => {
  const handleButtonClick = jest.fn()
  const component = renderer.create(
    <Hello buttonClicked={handleButtonClick} className={'btn-primary'} disabled>
      Hello Web Component Snapshot! ðŸ“¸
    </Hello>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
