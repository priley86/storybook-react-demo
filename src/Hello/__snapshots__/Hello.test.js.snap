// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello renders default properly 1`] = `
<hello-component
  class="btn-primary"
>
  Hello Web Component Snapshot! ðŸ“¸
</hello-component>
`;
